### ISSUE TITLE
Router error after redirect

### ISSUE THREAD

**Issue by jonathaneckman** [aurelia-contrib/aurelia-open-id-connect#47]
I've been tracing this error all week and havent made much progress so I'm hoping you'll see something I dont. After a login redirect, the following happens:

1. Auth server redirects back to /signin-oidc as expected. 
1. My root component begins to load (app.ts)
1. `OpenIdConnectNavigationStrategies.signInRedirectCallback` navigation strategy begins.
1. Within the aurelia router, something explodes. I believe it is [on this line](https://github.com/aurelia/router/blob/dffe90701cbd81ccd9fec2ab220cd8d1183a7e34/dist/es2015/aurelia-router.js#L1004) in `evaluateNavigationStrategy`. I see that `instruction.config.moduleId` is undefined.
1. The error below is thrown.
1. The page refreshes to redirect route specified in `OpenIdConnectConfiguration.loginRedirectRoute`.

```
ERROR [app-router] TypeError: Cannot read property 'trim' of undefined
ERROR [app-router] Router navigation failed, and no previous location or fallbackRoute could be restored.
```

So even though an error is thrown, things seem to be working as expected. Is this a known issue or is this a problem on my end? 

I have attempted to clone your plugin to troubleshoot further but get the following error whenever I `npm link` to this project. This is new for me so I may not be doing it right. I am going to keep trying so if you have any insight into how you run the plugin locally to test within a consuming app I'd appreciate it.  I think I've guessed the steps to build, run, and test but if you could document those as well it would be a huge help.

```
Error: Can't figure out a normalized module name for ./open-id-connect-user-block, please call PLATFORM.moduleName() somewhere to help.
    at Compilation.compilation.plugin.modules (C:\Code\myproject\node_modules\aurelia-webpack-plugin\dist\PreserveModuleNamePlugin.js:44:31)
```

**Comment by jonathaneckman** [comment]
I dont believe the `npm link` issue is caused by anything in this plugin. I reproduced similar issues with aurelia-validation. I [submitted an issue](https://github.com/aspnet/JavaScriptServices/issues/1453) to JavaScriptServices.

**Comment by jonathaneckman** [comment]
I stepped through this again. The problem is that the routes do not have a module id. I can see that when attempting to load the `logInRedirectCallback` navigation instruction, the `viewPortPlan.config.moduleId` is undefined. 

Starting in the navigation pipeline:
![image](https://user-images.githubusercontent.com/11893200/35189353-aed3af34-fe16-11e7-9a4a-5f1e2461b68f.png)

Then attempting to load the route:
![image](https://user-images.githubusercontent.com/11893200/35189349-973dbe28-fe16-11e7-833f-133c9cca3050.png)

Which tries to load a component based on the module id. In `aurelia-path` the error is thrown because of the call to `trim` an undefined parameter.

![image](https://user-images.githubusercontent.com/11893200/35189366-0bd1faba-fe17-11e7-9f4b-67fc2f45b57d.png)

I think the fix is simple. A module id needs to be added to the [login ](https://github.com/shaunluttin/aurelia-open-id-connect/blob/fd9321c7f30ab0ed0050677e2925cca5312c1f93/src/open-id-connect-routing.ts#L28) and [logout ](https://github.com/shaunluttin/aurelia-open-id-connect/blob/fd9321c7f30ab0ed0050677e2925cca5312c1f93/src/open-id-connect-routing.ts#L42) routes you are configuring in `configureRouter`.

**Comment by jonathaneckman** [comment]
I put together a [repro ](https://github.com/jonathaneckman/aurelia-router-repro). You can the issue in my [configureRouter](https://github.com/jonathaneckman/aurelia-router-repro/blob/bad18194f21c4b8f96f0a2cdf5950bec8de6eacf/src/app.ts#L31). 

Waiting to see if this is a bug in `aurelia-router` or if it should be a change here. See [this issue](https://github.com/aurelia/router/issues/289) there for more info.

**Comment by shaunluttin** [comment]
You did a ton a research looking in to that and even uploaded screenshots. Thank you for the report! 

I have been away from this project for about a month now and am not sure when I will be returning to it.

From quickly reviewing your analysis, it seems that the fix might be simple but also that it might be a bug in the Aurelia router. 

I would be happy to hear more about this and will check-in from time to time.

**Comment by jonathaneckman** [comment]
Sounds good. I have a feature and a blog post showcasing your plugin waiting on this issue so I'm motivated to help. If the fix is here, are you willing to accept a PR?

**Comment by shaunluttin** [comment]
Yes. I am definitely willing to accept a PR. 

Question: Have you checked out [the demo projects?](https://github.com/shaunluttin/aurelia-open-id-connect-demos) Is the issue there too?

**Comment by jonathaneckman** [comment]
Yes the issue is in the demo projects. 

![image](https://user-images.githubusercontent.com/11893200/35477390-64e5d4a0-038f-11e8-82b7-d9ad7af1442c.png)

**Comment by shaunluttin** [comment]
It occurred to me that the issue might be related to enhancement https://github.com/shaunluttin/aurelia-open-id-connect/issues/17 that happened with this commit https://github.com/shaunluttin/aurelia-open-id-connect/commit/11ea315a8ff741c670e1f33cd332a397aa87efbc.

**Comment by saspeed** [comment]
I was hoping to find some resolution to this issue this with the new release 2.0.0. 
But i can still confirm the problem even with the demo project. 
How is this not affecting more users of the plugin? 
@jonathaneckman Is there a workaround you can suggest?

**Comment by shaunluttin** [comment]
It's good to know that this remains a problem. I'll mark it as a priority; that said, I am not sure when I will have time to address it.

**Comment by rmja** [comment]
Hi,

I have looked into this and fount that the problem comes from the way redirects are performed for the signin/signout navigation strategies. This issue comes from the fix introduced by https://github.com/aurelia-contrib/aurelia-open-id-connect/commit/e06a1943e6a4688b0d90f5f8a332b8039a29a216 fixing https://github.com/aurelia-contrib/aurelia-open-id-connect/issues/46.

This is the current sequence when signing in:

1. signin-oidc route is hit from the idp.
2. window.location is assigned in the navigationStrategy callback registered for the route. No other fields are assigned on the instruction config which is required by the aurelia router.
3. _the router pipeline continues after the navigationStrategy callback has completed_, eventually failing with the 'trip' TypeError because the callback did not setup a valid configuration for the router to complete.
4. _The entire application is reloaded_ because of the assigned window.location. This time the assigned redirectRoute is loaded by the router and not the signin-oidc route.

Here is the desired sequence when signing in:
1. signin-oidc route is hit from the idp.
2. instruction.config.redirect is assigned in the navigationStrategy callback which is a valid configuration for the aurelia router.
2.1. I have been unable to reproduce #46, but if it is still an issue, rewrite the history using something like `instruction.router.history.navigate(redirectRoute, { replace: true });`
3. the router pipeline continues with a configured redirect, and the router perferms the redirect completing the login.

cc @shaunluttin

**Comment by shaunluttin** [comment]
We might have resolved this (at least partially) with v2.0.2.

**Comment by shaunluttin** [comment]
The fix is installed here: https://github.com/aurelia-contrib/aurelia-open-id-connect-demos.

**Comment by saspeed** [comment]
Awesome, Thanks! I'll verify next week

**Comment by rusch1981** [comment]
I believe there is a issue related the fixed installed (v2.0.2).

Running version 2.0.2 I am received the error:  
![image](https://user-images.githubusercontent.com/21145454/61801466-d47f7f00-adfc-11e9-81ce-a4e1949ea03d.png)

After debugging into aurelia-router.js, specifically the _buildNavigationPlan function, I noticed that when returning from identity server the parameter "instruction" (of type NavigationInstruction)  was coming through with a "config.redirect" property that contains a value.  The value is the url back to the last navigated location of the client application.  When _router.generate is called you get an error, because the "newinstruction.config.name" is undefined.  

![image](https://user-images.githubusercontent.com/21145454/61802246-2543a780-adfe-11e9-9379-020eb944f346.png)

When I remove the "config.redirect" property during debug, I recieved the trim error noted in the screen shot provided by @jonathaneckman.

I set the version back to 2.0.1 and I am no longer seeing any errors.  I hope that is helpful.
